presto-main/src/main/java/com/facebook/presto/operator/aggregation/AggregationMetadata.java

```diff
@@ -156,7 +160,9 @@ private static void verifyExactOutputFunction(MethodHandle method, Class<?> stat
             return;
         }
         Class<?>[] parameterTypes = method.type().parameterArray();
-        checkArgument(parameterTypes.length == 2 && parameterTypes[0] == stateInterface && parameterTypes[1] == BlockBuilder.class, "Output function must have the signature (%s, BlockBuilder)", stateInterface.getSimpleName());
+        checkArgument(parameterTypes.length >= 2, "Output function must take at least 2 arguments.");
+        checkArgument(Arrays.stream(parameterTypes).filter(type -> type.equals(stateInterface)).count() == 1, "Output function must take exactly one @AggregationState of type %s", stateInterface.getSimpleName());
+        checkArgument(Arrays.stream(parameterTypes).filter(type -> type.equals(BlockBuilder.class)).count() == 1, "Output function must take exactly one BlockBuilder parameter");
```

Comment:
I understand that this is after the method handle is bound. In this case, why would you have more than 2 arguments?